Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py

```diff
@@ -19,6 +26,243 @@ def makebuf(text):
     return BIO.MemoryBuffer(text)
 
 
+def create_pem_string(base64_cert: str) -> str:
+    """This function takes the base64 encoded certificate received via
+    ad-search in the 'UserCertificate' attribute and converts it to a PEM
+
+    Args:
+        base64_cert (str): certificate string with no BEGIN or END markers
+
+    Returns:
+        pemstring (str): PEM string for the certificate
+    """
+    pemstring = '-----BEGIN CERTIFICATE-----\n'
+    pemstring += '\n'.join(
+        [base64_cert[i: i + 64] for i in range(0, len(base64_cert), 64)]
+    )
+    pemstring += '\n-----END CERTIFICATE-----\n'
+    return pemstring
+
+
+def handle_attachment(attachment_part: Message) -> None:
+    """
+    Extracts and saves the attachment file to the context.
+    p7 files will be ignored.
+
+    Args:
+        attachment_part: attachment message part
+
+    """
+
+    fileName = attachment_part.get_filename()
+    if not fileName:
+        demisto.debug('Got nameless attachment, generating file name')
+        mime_type = attachment_part.get_content_type()
+        extension = mimetypes.guess_extension(mime_type) or '.bin'
+        fileName = f'attachment_{uuid.uuid4()}{extension}'
+
+    if (fileName.lower().endswith('.p7')
+            or fileName.lower().endswith('.p7', 0, -1)):
```

Comment:
The mime messages include a p7 file the message as an attachment, we don't need that file since we are already parsing the message to text. Without that condition that file will be extracted as if it was another attachment from the email